subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java

```diff
@@ -95,8 +95,7 @@
     void reject(String... versions);
 
     /**
-     * Rejects all versions of this component. Can be used to declare that a component is incompatible with another
-     * (typically, cannot have both a 2 different implementations of the same API).
+     * Rejects all versions of this component. Can be used to banish a specific component from dependency graphs.
```

Comment:
Does this "banish" a component (which sounds like an exclude) or make resolution fail if that component is in the graph?

Reply:
It makes resolution fail, I'll clarify.